src/userscores/view/exportdialogmodel.cpp

```diff
@@ -0,0 +1,473 @@
+
+    emit dataChanged(modelIndex, modelIndex);
+}
+
+void ExportDialogModel::setAllSelected(bool selected)
+{
+    QModelIndexList previousSelectedIndexes = m_selectionModel->selectedIndexes();
+    m_selectionModel->QItemSelectionModel::select(QItemSelection(index(0), index(m_notations.size() - 1)),
+                                                  selected ? QItemSelectionModel::Select : QItemSelectionModel::Deselect);
+    QModelIndexList newSelectedIndexes = m_selectionModel->selectedIndexes();
+
+    QSet<QModelIndex> indexesToUpdate(previousSelectedIndexes.begin(), previousSelectedIndexes.end());
+    indexesToUpdate = indexesToUpdate.unite(QSet<QModelIndex>(newSelectedIndexes.begin(), newSelectedIndexes.end()));
+
+    for (const QModelIndex& indexToUpdate : indexesToUpdate) {
+        emit dataChanged(indexToUpdate, indexToUpdate);
+    }
+}
+
+void ExportDialogModel::selectCurrentNotation()
+{
+    for (int i = 0; i < m_notations.size(); i++) {
+        setSelected(i, m_notations[i] == context()->currentNotation());
+    }
+}
+
+IMasterNotationPtr ExportDialogModel::masterNotation() const
+{
+    return context()->currentMasterNotation();
+}
+
+bool ExportDialogModel::isMainNotation(INotationPtr notation) const
+{
+    return notation == masterNotation()->notation();
+}
+
+bool ExportDialogModel::isNotationIndexValid(int index) const
+{
+    return index >= 0 && index < m_notations.size();
+}
+
+int ExportDialogModel::selectionLength() const
+{
+    return m_selectionModel->selectedIndexes().size();
+}
+
+QString ExportDialogModel::selectedExportSuffix()
+{
+    return m_selectedExportSuffix;
+}
+
+void ExportDialogModel::setExportSuffix(QString suffix)
+{
+    if (m_selectedExportSuffix == suffix) {
+        return;
+    }
+
+    m_selectedExportSuffix = suffix;
+    emit selectedExportSuffixChanged(suffix);
+
+    setUnitType(static_cast<int>(exportScoreService()->supportedUnitTypes(suffix.toStdString())[0]));
+}
+
+QList<int> ExportDialogModel::availableUnitTypes() const
+{
+    QList<int> result;
+    for (notation::INotationWriter::UnitType t : exportScoreService()->supportedUnitTypes(m_selectedExportSuffix.toStdString())) {
+        result << static_cast<int>(t);
+    }
+    return result;
+}
+
+int ExportDialogModel::selectedUnitType() const
+{
+    return static_cast<int>(m_selectedUnitType);
+}
+
+void ExportDialogModel::setUnitType(int unitType)
+{
+    setUnitType(static_cast<ExportUnitType>(unitType));
+}
+
+void ExportDialogModel::setUnitType(ExportUnitType unitType)
+{
+    if (m_selectedUnitType == unitType) {
+        return;
+    }
+
+    m_selectedUnitType = unitType;
+    emit selectedUnitTypeChanged(unitType);
+}
+
+bool ExportDialogModel::exportScores()
+{
+    INotationPtrList notations;
+
+    for (const QModelIndex& index: m_selectionModel->selectedIndexes()) {
+        notations.push_back(m_notations[index.row()]);
+    }
+
+    IF_ASSERT_FAILED(!notations.empty()) {
```

Comment:
we can call the export score method from qml if we did not select any scores?

Reply:
The button is (should be) disabled when no scores are selected. But I will change into a normal `if`.